test/catalog_test.exs

```diff
@@ -21,7 +21,7 @@ defmodule Wanda.CatalogTest do
         |> Enum.sort()
         |> Enum.filter(fn file -> file != "malformed_check.yaml" end)
 
-      catalog = Catalog.get_catalog()
+      catalog = Catalog.get_catalog(%{"provider" => "azure"})
```

Comment:
I think the previous implementation made more sense, because it was testing that the catalog is retrieved entirely if no `env` is given, which is something we should still be testing.
More when the test name is not changed.
I would keep this as it was and create new tests with `azure` as provider

Reply:
> that the catalog is retrieved entirely if no env is given

Good catch, it actually makes sense to return the whole catalog if no env is given. Current PR doesn't work that way.